### PR TITLE
Python 3.12 support

### PR DESCRIPTION
**Pull request by tnoczyns-volue**
Support Python 3.12.
Drop Python 3.8.

### CODE REVIEW

**Comment by tnoczyns-volue** [comment]
**TL:DR**

Support for Python 3.12 is doable 😄 
More info below.

### Windows - user

Works out of the box.

### Ubuntu (20.04) - user

Needed to use updated:
* setuptools - tested with latest one 69.1.0, which has Python>=3.8
* six - works only with the latest one 1.16.0
https://bugzilla.mozilla.org/show_bug.cgi?id=1857492

### Windows - developer

Problem with Numpy. With Python>=3.8 set in package target versions, Numpy max version 1.24.4 is installed. It does not support Python 3.12.

Two possible solutions:
* Release Mesh Python SDK 1.5 version with current changes, without explicit Python 3.12 support and add Python 3.8 deprecation warning. Then remove support for Python 3.8, add Python 3.12, and immediately release 1.6[^1].
* Try to support both Python 3.8 and 3.12 in Mesh Python SDK 1.5 version by adding explicit numpy dependency to the project and use [multiple constraints dependencies](https://python-poetry.org/docs/dependency-specification/#multiple-constraints-dependencies).  Haven't tested this approach yet.

[^1]: Assuming support for Python 3.12 is required by some customer.

**Comment by tnoczyns-volue** [comment]
After discussion with @torsannes we are going with option number 1:
> Release Mesh Python SDK 1.5 version with current changes, without explicit Python 3.12 support and add Python 3.8 deprecation warning. Then remove support for Python 3.8, add Python 3.12, and immediately release 1.6.

BTW Python 3.8 end-of-life is scheduled for October 2024. See: https://devguide.python.org/versions/#supported-versions.

File: .github/workflows/install_dev_env/action.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ runs:
     - name: Installing poetry
       uses: abatilo/actions-poetry@v2.0.0
       with:
-        poetry-version: '1.2.2'
+        poetry-version: '1.7.1'
```
Comment (tnoczyns-volue) [inline .github/workflows/install_dev_env/action.yml:20]: Poetry 1.7.0 has added support for Python 3.12. 
See: https://python-poetry.org/blog/announcing-poetry-1.7.0/#official-support-for-python-312

**Review by erny-powel — APPROVED** [review]
🎉
